### PR TITLE
Feature/world box ori zxy fix

### PR DESCRIPTION
**Pull request by yappel**
The euler angles that get added to the marker sensor get converted correctly

### CODE REVIEW

File: IRescue/Core/DataTypes/RotationMatrix.cs [hunk 1]
```diff
@@ -90,5 +95,34 @@ private static float[] CreateMatrixVector(float xRotation, float yRotation, floa
                 };
             return rot;
         }
+
```
Comment (remcodevos) [inline IRescue/Core/DataTypes/RotationMatrix.cs]: These two private methods have no calling hierachy, is there a specific reason for them to be here for feature use or should they be removed?

**Comment by remcodevos** [comment]
I added some missing tests in TransformationMatrix and MarkerSensor. In RotationMatrix a test is missing for creation from a quaternion and I wasn't sure what to do with that.

One issue is that TestReadMarkerSensorData fails in ParticleFilterTest. All the code looks good.

File: IRescue/Core/DataTypes/RotationMatrix.cs (hunk 1 above)
Comment (yappel) [inline IRescue/Core/DataTypes/RotationMatrix.cs]: CreateMatrixVector(float xRotation, float yRotation, float zRotation)
Gets called in the constructor which accepts euler angles. That constructor does not get called in the current version of the unity code, but has been called a lot in the past and might be called again in the future.
CreateMatrixVector(Vector3 axis, float angle)
Does not get used anywhere anymore and can get removed indeed.

**Comment by yappel** [comment]
On my machine all tests succeed so what is the message when the test fails on your machine?

About the test creating a RotationMatrix from quaternion. I get 100% coverage on the RotationMatrix class when only running the RotationMatrixTest tests. There is a test for creating a rotation matrix from quaternion right here : https://github.com/yappel/BEP_meta1/blob/feature/world_box_ori_zxy_fix/IRescue/Core/Tests/Core.Test/DataTypes/RotationMatrixTest.cs#L51

**Comment by lkrombeen** [comment]
Unity test is failing
image

**Comment by lkrombeen** [comment]
Fixed the test, markersensor is an orientation source but this was not changed in the tests.

**Pull request merged**